app/lib/actions/foreman_ansible/play_hostgroup_roles.rb

```diff
@@ -0,0 +1,65 @@
+module Actions
+  module ForemanAnsible
+    # Actions that initiaztes the playbook run for roles assigned to
+    # the host. It doest that either locally or via a proxy when available.
+    class PlayHostgroupRoles < Actions::EntryAction
+      include ::Actions::Helpers::WithContinuousOutput
+      include ::Actions::Helpers::WithDelegatedAction
+
+      def plan(hostgroup, proxy_selector = ::ForemanAnsible::ProxySelector.new)
+        if hostgroup.hosts.empty?
+          raise ::Foreman::Exception.new(N_('hostgroup is empty'))
+        end
+        input[:hostgroup] = { :id => hostgroup.id, :name => hostgroup.name }
+        proxy = proxy_selector.determine_proxy(hostgroup.hosts[0])
```

Comment:
I just want to make clear that we want to play roles on a group of hosts with Ansible, like:

```
PLAY [all] *********************************************************************

TASK [install : run installer] ***

ok: [node01]
ok: [node02]
```

Starting a single play for each host is not wanted because it implies several problems. Basically, the control over the orchestration gets lost. One problem is that each host proceeds in its own tempo, but this is actually what [Ansible strategies](http://docs.ansible.com/ansible/playbooks_strategies.html) are responsible for. Another problem is that a failure of one host does not stop the entire provisioning process and leaves the hostgroup in an uncontrolled state. Sometimes, timing between the tasks is essential and Ansible tasks strictly need to be executed linearly. 
 
I also see that you are a bit concerned with the implementation here because it does not honour the concept of smart proxies well. However, I don't know how a better solution should look like and if the playbook fails because hosts are unreachable, I think is reasonable. With this background, do you have another idea of how to do this better?